### PR TITLE
Create use case for take turn.

### PR DESCRIPTION
**Pull request by minaccchiang**
Need two more method for presenter.

waiting for the change of place word use case 

fixed: GameState is static issue

### CODE REVIEW

File: src/main/java/Take_Turn/TakeTrunInputData.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    private final Coordinate end;
+    private final Player currplayer;
+    private final GameState gameState;
+
+    public TakeTrunInputData(String word, Coordinate start, Coordinate end, Player currplayer, GameState gameState) {
+        this.word = word;
+        this.start = start;
+        this.end = end;
+        this.currplayer = currplayer;
+        this.gameState = gameState;
```
Comment (JosephSun2003) [inline src/main/java/Take_Turn/TakeTrunInputData.java:20]: Isn't requried since gameState is static.

**Review by JosephSun2003 — CHANGES_REQUESTED** [review]
Overall, everything should work, a few changes to references to gameState since we can call it from as GameState.methodName();

**Comment by JosephSun2003** [comment]
MERGE ONLY AFTE HRID's Changes are MERGED.

**Review by JosephSun2003 — DISMISSED** [review]
Changes look good. No concerns noticed.

**Review by ethanhonest — COMMENTED** [review]
The inputData should be passed as parameter to taketurn() as opposed to in the constructor. I will create the constructor in main, so I won't know what to pass in until the user manually puts in their instructions. So, please remove inputData as a parameter from the constructor, and add it as a parameter to taketurn().

Otherwise, looks good, and I will approve once that change is made

**Review by JosephSun2003 — APPROVED** [review]
Code looks good, aside from Ethan's comment, everything looks fine.

**Comment by JosephSun2003** [comment]
Resolved the conflicts with main, mostly minor things.

**Review by JosephSun2003 — APPROVED** [review]
Overall, merging went nicely since most of the code were repeats of one another.

**Review by ethanhonest — APPROVED** [review]
Looks like there are minor typos in the names of the files, but overall the code looks good

**Review by patelhrid — APPROVED** [review]
Looks good to me.

**Pull request merged**